app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt

```diff
@@ -290,6 +264,44 @@ class ParkingViewModel(
     }
   }
 
+  fun handleReviewDeletion(parking: Parking = selectedParking.value!!, newScore: Double) {
+    parking.avgScore =
+        if (parking.nbReviews >= 2) {
+          ((parking.nbReviews * parking.avgScore) - newScore) / (parking.nbReviews - 1)
+        } else {
+          0.0
+        }
+    parking.nbReviews -= 1
+    parkingRepository.updateParking(parking, {}, {})
+  }
+
+  fun handleNewReview(parking: Parking = selectedParking.value!!, newScore: Double) {
+    parking.avgScore =
+        (100 * ((parking.avgScore * parking.nbReviews) + newScore) / (parking.nbReviews + 1))
+            .toInt() / 100.00
+    parking.nbReviews += 1
+    parkingRepository.updateParking(
+        parking,
+        onSuccess = {
+          val tile = Tile.getTileFromPoint(parking.location.center)
+          tilesToParking.value[tile]?.let { parkingList ->
+            tilesToParking.value[tile] =
+                parkingList.map { p -> if (p.uid == parking.uid) parking else p }
+          }
+        },
+        onFailure = {})
+  }
+
+  fun handleReviewUpdate(
+      parking: Parking = selectedParking.value!!,
+      newScore: Double,
+      oldScore: Double
+  ) {
+    val delta = if (parking.nbReviews != 0) (newScore - oldScore) / parking.nbReviews else 0.0
```

Comment:
We shouldn't `if else` on the nbReviews. 
If it is zero and the user can modify his review, something has gone. 
Making the app crash with an exception might be overkill, but using `Log.e` seems reasonable and then instantly returning without running the code

Reply:
I agree here, but not in the deletion logic. In that one, the number of parkings being 0 makes sense. I will update here though